homework/minieieva_daria/report.md

```diff
@@ -0,0 +1,100 @@
+### Week 6 Regular Expressions Homework
+
+### This is a template for the report of 21 problems on [this site](http://regextutorials.com/excercise.html).
+
+### Fork this repository, change this file and send us the link for grading.
+
+### We recommend you (re)watch our presentation, and use a [cheatsheet](./cheatsheet.md) while working on the following problems.
+
+### If you are having problems with figuring out the correct solution, skip it for later, and if you are completely lost, use the hints and try to understand them.
+
+---
+
+### Problem 1: Floating point numbers
+```
+\d\.\d+
+```
+### Problem 2: Years before 1990
+```
+\d\s\w*\s\w*[,|:]*(\s\w*)*\(1[0-9][0-8][0-9]\)
+```
+### Problem 3: Hexadecimal colors
+```
+#([A-Z]|[0-9]){6}
+```
+### Problem 4: Grayscale colors
+```
+#((([A-z]|[0-9]){1,2})\2{2})
+```
+### Problem 5: Too long lines
+```
+.{30,}
```

Comment:
Yup, pretty basic here. If you are interested, there is a nice read on why [Why Using the Greedy .* in Regular Expressions Is Almost Never What You Actually Want](https://mariusschulz.com/blog/why-using-the-greedy-in-regular-expressions-is-almost-never-what-you-actually-want), doesn't change anything in this particular problem but is a nice dive into the complexities of regular expressions!